include/ilang/ila/validate_model.h

```diff
@@ -0,0 +1,22 @@
+#ifndef ILANG_ILA_VALIDATE_MODEL_H__
+#define ILANG_ILA_VALIDATE_MODEL_H__
+
+#include <ilang/ila/ast_fuse.h>
```

Comment:
Seems like `ast_fuse.h` and `z3_expr_adapter.h` are not used/required in the header. Implementation-only usage should be included in the source files.